driver-core/src/main/com/mongodb/internal/binding/SingleServerBinding.java

```diff
@@ -79,6 +81,11 @@ public ServerApi getServerApi() {
         return serverApi;
     }
 
+    @Override
+    public RequestContext getRequestContext() {
+        return IgnorableRequestContext.INSTANCE;
```

Comment:
Instances of `SingleServerBinding` are crated only from `MongoClient.cleanCursors`, which is in turn called only from a background service thread. While it is clear for the creator of `SingleServerBinding` that using `IgnorableRequestContext.INSTANCE` is fine, it is not clear from inside the `SingleServerBinding` class. I think this class should have `RequestContext` as a constructor parameter, and then `MongoClient` should supply `IgnorableRequestContext.INSTANCE` when calling the constructor.